src/pages/index.js

```diff
@@ -69,15 +37,59 @@ const profileDescriptionInput = document.querySelector(
   "#profile-description-input"
 );
 
-// Cards
-const addCardModal = document.querySelector("#add-card-modal");
+// ********************************* API ******************************** //
 
-// Modals
-const profileEditForm = profileEditModal.querySelector(".modal__form");
-const addButtonForm = addCardModal.querySelector(".modal__form");
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "16b910f9-e28d-483c-ac51-2ec35ebc8735",
+    "Content-Type": "application/json",
+  },
+});
+
+let userInfoInstance;
+
+// Initialiize, Generate User Profile and Cards on Load
+api
+  .getUserInfo()
+  .then((userInfo) => {
+    const { name, about, avatar } = userInfo;
+    userInfoInstance = new UserInfo(
+      profileTitle,
+      profileDescription,
+      avatarImg
+    );
+
+    userInfoInstance.setUserInfo({
+      name,
+      job: about,
+      avatar: avatar,
+    });
+    api.getInitialCards().then((res) => {
+      console.log(res);
+      res.forEach((item) => {
+        renderCard(item);
+      });
+    });
+  })
+  .catch((error) => {
+    console.log(error);
+  });
 
-// User Info Instantiation
-const userInfo = new UserInfo(profileTitle, profileDescription);
+// Edit User Profile
+const apiUpdateProfileInfo = (newInfo) => {
+  return api
+    .editUserInfo(newInfo)
+    .then((updatedInfo) => {
+      profileTitleInput.value = updatedInfo.name;
+      profileDescriptionInput.value = updatedInfo.about;
```

Comment:
```js

      profileTitleInput.value = updatedInfo.name;
      profileDescriptionInput.value = updatedInfo.about;
```


This is needed only when you **open** the profile modal. No need to make api calls in that case. It's a waste of resources. You have `getUserInfo` in class `UserInfo` for that